internal/services/worker.go

```diff
@@ -458,7 +458,16 @@ func (w *WorkerPool) processCreateInstanceTask(ctx context.Context, task *models
 			// Should not happen if hosts were provided and valid, but handle defensively
 			logger.Warnf("Worker: Inventory path empty for instance ID %d, skipping playbook run", instance.ID)
 		} else {
-			if err := provisioner.RunAnsiblePlaybook(inventoryPath); err != nil {
+			// tags depending on the provisioner
+			tags := []string{}
+			if instanceReq.Provider == models.ProviderXimera {
+				tags = []string{"setup"}
+			}
+			if instanceReq.Provider == models.ProviderDO {
+				tags = []string{"setup", "volumes"}
+			}
+
+			if err := provisioner.RunAnsiblePlaybook(inventoryPath, tags); err != nil {
```

Comment:
Q: why is tag modified here?

Reply:
Because in ximera, we don't need to run the `volumes` playbook, as we don't need to persist external volumes, as we attach the disk directly during creation.
In order to select what playbooks to run, I needed to modify this part.